cf_xarray/tests/test_parametric.py

```diff
@@ -0,0 +1,613 @@
+        [
+            [
+                [
+                    [1.0, 3.0, 5.0],
+                    [1.0, 3.0, 5.0],
+                ],
+                [
+                    [1.0, 3.0, 5.0],
+                    [1.0, 3.0, 5.0],
+                ],
+            ],
+            [
+                [
+                    [1.0, 3.0, 5.0],
+                    [1.0, 3.0, 5.0],
+                ],
+                [
+                    [1.0, 3.0, 5.0],
+                    [1.0, 3.0, 5.0],
+                ],
+            ],
+        ],
+        dims=("time", "lat", "lon", "lev"),
+        name="z",
+    )
+
+    assert_allclose(output, expected)
+
+    assert output.attrs["standard_name"] == "altitude"
+
+
+def test_ocean_s_coordinate_g2():
+    C = xr.DataArray([0, 1, 2], dims=("lev",), name="C")
+
+    transform = parametric.OceanSG2.from_terms(
+        {
+            "s": s,
+            "c": C,
+            "eta": eta,
+            "depth": depth,
+            "depth_c": depth_c,
+        }
+    )
+
+    output = transform.decode()
+
+    expected = xr.DataArray(
+        [
+            [
+                [
+                    [1.0, 3.0, 5.0],
+                    [1.0, 3.0, 5.0],
+                ],
+                [
+                    [1.0, 3.0, 5.0],
+                    [1.0, 3.0, 5.0],
+                ],
+            ],
+            [
+                [
+                    [1.0, 3.0, 5.0],
+                    [1.0, 3.0, 5.0],
+                ],
+                [
+                    [1.0, 3.0, 5.0],
+                    [1.0, 3.0, 5.0],
+                ],
+            ],
+        ],
+        dims=("time", "lat", "lon", "lev"),
+        name="z",
+    )
+
+    assert_allclose(output, expected)
+
+    assert output.attrs["standard_name"] == "altitude"
+
+
+def test_ocean_sigma_z_coordinate():
+    zlev = xr.DataArray(
+        [0, 1, np.nan], dims=("lev",), name="zlev", attrs={"standard_name": "altitude"}
+    )
+
+    _sigma = xr.DataArray([np.nan, np.nan, 3], dims=("lev",), name="sigma")
+
+    transform = parametric.OceanSigmaZ.from_terms(
+        {
+            "sigma": _sigma,
+            "eta": eta,
+            "depth": depth,
+            "depth_c": depth_c,
+            "nsigma": 10,
+            "zlev": zlev,
+        }
+    )
+
+    output = transform.decode()
+
+    expected = xr.DataArray(
+        [
+            [
+                [[0.0, 0.0], [0.0, 0.0]],
+                [[0.0, 0.0], [0.0, 0.0]],
+            ],
+            [
+                [[1.0, 1.0], [1.0, 1.0]],
+                [[1.0, 1.0], [1.0, 1.0]],
+            ],
+            [
+                [[7.0, 7.0], [7.0, 7.0]],
+                [[7.0, 7.0], [7.0, 7.0]],
+            ],
+        ],
+        dims=("lev", "time", "lat", "lon"),
+        name="z",
+    )
+
+    assert_allclose(output, expected)
+
+    assert output.attrs["standard_name"] == "altitude"
+
+
+def test_ocean_double_sigma_coordinate():
+    k_c = xr.DataArray(
```

Comment:
shouldn't these be scalars? How are they stored in the netCDF files you're looking at?

Reply:
@dcherian See comment above, changed all the constants over and verified that's how they're stored in netCDF files.